File: src/main/java/toast/pine/Adapters/ItemAdapter.java
package toast.pine.Adapters;

import org.bukkit.persistence.PersistentDataAdapterContext;
import org.bukkit.persistence.PersistentDataType;
import org.jetbrains.annotations.NotNull;
import toast.pine.Classes.Items.Item;

public class ItemAdapter implements PersistentDataType<Item, Item> {

    @Override
    public @NotNull Class<Item> getPrimitiveType() {
        return Item.class;
    }

    @Override
    public @NotNull Class<Item> getComplexType() {
        return Item.class;
    }

    @Override
    public @NotNull Item toPrimitive(@NotNull Item item, @NotNull PersistentDataAdapterContext persistentDataAdapterContext) {
        return item;
    }

    @Override
    public @NotNull Item fromPrimitive(@NotNull Item item, @NotNull PersistentDataAdapterContext persistentDataAdapterContext) {
        return item;
    }
}

File: src/main/java/pine/toast/toastrpg/adapters/ItemAdapter.kt
package pine.toast.toastrpg.adapters

import org.bukkit.persistence.PersistentDataAdapterContext
import org.bukkit.persistence.PersistentDataType
import pine.toast.toastrpg.classes.items.Item


class ItemAdapter : PersistentDataType<Item, Item> {
    @get:JvmName("getPrimitiveTypeItem")
    private val primitiveType: Class<Item>
        get() = Item::class.java
    @get:JvmName("getComplexTypeItem")
    private val complexType: Class<Item>
        get() = Item::class.java

    override fun toPrimitive(item: Item, persistentDataAdapterContext: PersistentDataAdapterContext): Item {
        return item
    }

    override fun fromPrimitive(item: Item, persistentDataAdapterContext: PersistentDataAdapterContext): Item {
        return item
    }

    override fun getPrimitiveType(): Class<Item> {
        return primitiveType
    }

    override fun getComplexType(): Class<Item> {
        return complexType
    }
}
